exercises/practice/dnd-character/dnd-character_spec.lua

```diff
@@ -69,121 +69,71 @@ describe('dnd', function()
-      assert.equal("integer", math.type(dexterity))
-      assert.lteq(3, dexterity)
-      assert.lteq(dexterity, 18)
-      assert.equal(dexterity, character.dexterity)
-    end)
+      assert.lteq(3, character.strength)
+      assert.lteq(character.strength, 18)
 
-    it('creates a character with valid constitution', function()
-      local character = dnd.Character:new("Charlie")
-      local constitution = character.constitution
-      assert.equal("integer", math.type(constitution))
-      assert.lteq(3, constitution)
-      assert.lteq(constitution, 18)
-      assert.equal(constitution, character.constitution)
-    end)
+      assert.lteq(3, character.dexterity)
+      assert.lteq(character.dexterity, 18)
 
-    it('creates a character with valid intelligence', function()
-      local character = dnd.Character:new("David")
-      local intelligence = character.intelligence
-      assert.equal("integer", math.type(intelligence))
-      assert.lteq(3, intelligence)
-      assert.lteq(intelligence, 18)
-      assert.equal(intelligence, character.intelligence)
-    end)
+      assert.lteq(3, character.constitution)
+      assert.lteq(character.constitution, 18)
 
-    it('creates a character with valid wisdom', function()
-      local character = dnd.Character:new("Eve")
-      local wisdom = character.wisdom
-      assert.equal("integer", math.type(wisdom))
-      assert.lteq(3, wisdom)
-      assert.lteq(wisdom, 18)
-      assert.equal(wisdom, character.wisdom)
-    end)
+      assert.lteq(3, character.intelligence)
+      assert.lteq(character.intelligence, 18)
 
-    it('creates a character with valid charisma', function()
-      local character = dnd.Character:new("Fred")
-      local charisma = character.charisma
-      assert.equal("integer", math.type(charisma))
-      assert.lteq(3, charisma)
-      assert.lteq(charisma, 18)
-      assert.equal(charisma, character.charisma)
+      assert.lteq(3, character.wisdom)
+      assert.lteq(character.wisdom, 18)
+
+      assert.lteq(3, character.charisma)
+      assert.lteq(character.charisma, 18)
+
+      assert.equal(10 + dnd.modifier(character.constitution), character.hitpoints)
     end)
 
-    it('creates a character with valid hitpoints', function()
-      for i = 1, 10 do
-        local character = dnd.Character:new(tostring(i))
-        assert.equal(10 + dnd.modifier(character.constitution), character.hitpoints)
-      end
+    it('each ability is calculated once', function()
+      local character = dnd.Character:new('Bob')
+
+      assert.equal(character.strength, character.strength)
+      assert.equal(character.dexterity, character.dexterity)
+      assert.equal(character.constitution, character.constitution)
+      assert.equal(character.intelligence, character.intelligence)
+      assert.equal(character.wisdom, character.wisdom)
+      assert.equal(character.charisma, character.charisma)
+      assert.equal(character.hitpoints, character.hitpoints)
```

Comment:
This doesn't make a ton of sense here unless these are computed properties (ie: using `__index`), but now we match the test specification.

Reply:
I'm sure it's meant to test that the value doesn't change between calls.